services/apps/alcs/src/portal/notice-of-intent-submission/notice-of-intent-owner/notice-of-intent-owner.service.spec.ts

```diff
@@ -0,0 +1,341 @@
+          uuid: parcelUuid,
+        }),
+      ],
+    });
+    mockRepo.findOneOrFail.mockResolvedValue(owner);
+    mockRepo.save.mockResolvedValue(new NoticeOfIntentOwner());
+
+    await service.removeFromParcel('', parcelUuid);
+
+    expect(owner.parcels.length).toEqual(0);
+    expect(mockRepo.findOneOrFail).toHaveBeenCalledTimes(1);
+    expect(mockRepo.save).toHaveBeenCalledTimes(1);
+  });
+
+  it('should set properties and call save for update', async () => {
+    const owner = new NoticeOfIntentOwner({
+      firstName: 'Bruce',
+      lastName: 'Wayne',
+    });
+    mockRepo.findOneOrFail.mockResolvedValue(owner);
+    mockRepo.save.mockResolvedValue(new NoticeOfIntentOwner());
+
+    await service.update('', {
+      firstName: 'I Am',
+      lastName: 'Batman',
+      email: '',
+      phoneNumber: '',
+      typeCode: '',
+    });
+
+    expect(owner.firstName).toEqual('I Am');
+    expect(owner.lastName).toEqual('Batman');
+    expect(mockRepo.findOneOrFail).toHaveBeenCalledTimes(1);
+    expect(mockRepo.save).toHaveBeenCalledTimes(1);
+  });
+
+  it('should delete the existing document when updating', async () => {
+    const owner = new NoticeOfIntentOwner({
+      firstName: 'Bruce',
+      lastName: 'Wayne',
+      corporateSummaryUuid: 'oldUuid',
+      corporateSummary: new NoticeOfIntentDocument(),
+    });
+    mockRepo.findOneOrFail.mockResolvedValue(owner);
+    mockRepo.save.mockResolvedValue(new NoticeOfIntentOwner());
+    mockAppDocumentService.delete.mockResolvedValue({} as any);
+
+    await service.update('', {
+      organizationName: '',
+      email: '',
+      phoneNumber: '',
+      typeCode: '',
+      corporateSummaryUuid: 'newUuid',
+    });
+
+    expect(owner.corporateSummaryUuid).toEqual('newUuid');
+    expect(mockAppDocumentService.delete).toHaveBeenCalledTimes(1);
+    expect(mockRepo.findOneOrFail).toHaveBeenCalledTimes(1);
+    expect(mockRepo.save).toHaveBeenCalledTimes(2);
+  });
+
+  it('should call through for delete', async () => {
+    mockRepo.remove.mockResolvedValue({} as any);
+
+    await service.delete(new NoticeOfIntentOwner());
+
+    expect(mockRepo.remove).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call through for verify', async () => {
+    mockRepo.findOneOrFail.mockResolvedValue(new NoticeOfIntentOwner());
+
+    await service.getOwner('');
+
+    expect(mockRepo.findOneOrFail).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call through for getMany', async () => {
+    mockRepo.find.mockResolvedValue([new NoticeOfIntentOwner()]);
+
+    await service.getMany([]);
+
+    expect(mockRepo.find).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call through for save', async () => {
+    mockRepo.save.mockResolvedValue(new NoticeOfIntentOwner());
+
+    await service.save(new NoticeOfIntentOwner());
+
+    expect(mockRepo.save).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call update for the application with the first parcels last name', async () => {
```

Comment:
'should call update for the noi with the first parcels last name'

Reply:
Fixed